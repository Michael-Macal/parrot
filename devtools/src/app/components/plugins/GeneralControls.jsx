import React, { Component } from 'react';
import PropTypes from 'prop-types';

import TextField from 'material-ui/TextField';
import FlatButton from 'material-ui/FlatButton';
import RefreshIcon from 'material-ui/svg-icons/navigation/refresh';

const textareaStyle = {
  paddingLeft: 0,
};

class GeneralControls extends Component {
  static proptypes = {
    url: PropTypes.string.isRequired,
    setUrl: PropTypes.func.isRequired,
    onRefresh: PropTypes.func.isRequired,
  }

  setUrl = (event, url) => this.props.setUrl(url);

  render() {
    const { url, onRefresh } = this.props;
    return (
      <div>
        <TextField
          name="Parrot Server Host"
          floatingLabelText="Parrot Server Host"
          value={url}
          onChange={this.setUrl}
          textareaStyle={textareaStyle}
        />
        <FlatButton
          icon={<RefreshIcon />}
          onTouchTap={onRefresh}
        />
      </div>
    );
  }
}


export default GeneralControls;
