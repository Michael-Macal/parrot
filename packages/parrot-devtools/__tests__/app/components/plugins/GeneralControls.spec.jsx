import React from 'react';
import { shallow } from 'enzyme';

import GeneralControls from '../../../../src/app/components/plugins/GeneralControls';

describe('GeneralControls component', () => {
  it('renders', () => {
    const props = {
      url: 'http://localhost:8080',
      setUrl: jest.fn(),
      onRefresh: jest.fn(),
    };
    const wrapper = shallow(<GeneralControls {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
