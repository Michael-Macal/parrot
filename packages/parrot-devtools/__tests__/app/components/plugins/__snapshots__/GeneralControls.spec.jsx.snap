// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralControls component renders 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Parrot Server Host"
      fullWidth={false}
      multiLine={false}
      name="Parrot Server Host"
      onChange={[Function]}
      rows={1}
      textareaStyle={
        Object {
          "paddingLeft": 0,
        }
      }
      type="text"
      underlineShow={true}
      value="http://localhost:8080"
    />
    <FlatButton
      disabled={false}
      fullWidth={false}
      icon={<NavigationRefresh />}
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      primary={false}
      secondary={false}
    />
  </div>,
  "nodes": Array [
    <div>
      <TextField
        disabled={false}
        floatingLabelFixed={false}
        floatingLabelText="Parrot Server Host"
        fullWidth={false}
        multiLine={false}
        name="Parrot Server Host"
        onChange={[Function]}
        rows={1}
        textareaStyle={
          Object {
            "paddingLeft": 0,
          }
        }
        type="text"
        underlineShow={true}
        value="http://localhost:8080"
      />
      <FlatButton
        disabled={false}
        fullWidth={false}
        icon={<NavigationRefresh />}
        labelPosition="after"
        labelStyle={Object {}}
        onKeyboardFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        primary={false}
        secondary={false}
      />
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <GeneralControls
        onRefresh={[Function]}
        setUrl={[Function]}
        url="http://localhost:8080"
      />,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": GeneralControls {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onRefresh": [Function],
          "setUrl": [Function],
          "url": "http://localhost:8080",
        },
        "refs": Object {},
        "setUrl": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Parrot Server Host"
            fullWidth={false}
            multiLine={false}
            name="Parrot Server Host"
            onChange={[Function]}
            rows={1}
            textareaStyle={
              Object {
                "paddingLeft": 0,
              }
            }
            type="text"
            underlineShow={true}
            value="http://localhost:8080"
          />
          <FlatButton
            disabled={false}
            fullWidth={false}
            icon={<NavigationRefresh />}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            primary={false}
            secondary={false}
          />
        </div>,
        "_renderedOutput": <div>
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="Parrot Server Host"
            fullWidth={false}
            multiLine={false}
            name="Parrot Server Host"
            onChange={[Function]}
            rows={1}
            textareaStyle={
              Object {
                "paddingLeft": 0,
              }
            }
            type="text"
            underlineShow={true}
            value="http://localhost:8080"
          />
          <FlatButton
            disabled={false}
            fullWidth={false}
            icon={<NavigationRefresh />}
            labelPosition="after"
            labelStyle={Object {}}
            onKeyboardFocus={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            onTouchTap={[Function]}
            primary={false}
            secondary={false}
          />
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <GeneralControls
    onRefresh={[Function]}
    setUrl={[Function]}
    url="http://localhost:8080"
  />,
}
`;
